action/protocol/poll/governance_protocol.go

```diff
@@ -239,7 +239,11 @@ func (p *governanceChainCommitteeProtocol) CalculateCandidatesByHeight(ctx conte
 func (p *governanceChainCommitteeProtocol) DelegatesByEpoch(ctx context.Context, epochNum uint64) (state.CandidateList, error) {
 	bcCtx := protocol.MustGetBlockchainCtx(ctx)
 	rp := rolldpos.MustGetProtocol(bcCtx.Registry)
-	tipEpochNum := rp.GetEpochNum(bcCtx.Tip.Height)
+	stateTipHeight, err := p.sr.Height()
```

Comment:
you are going to replace p.sr with sr from passed-in parameter, right?

Reply:
yes in next PR!